src/clojure_lsp/crawler.clj

```diff
@@ -167,12 +168,56 @@
     (concat (diagnose-unused-aliases uri declared-aliases unused-aliases)
             (diagnose-unused-references uri declared-references all-envs))))
 
-(defn find-diagnostics [project-aliases uri usages]
-  (let [unknown (diagnose-unknown project-aliases usages)
+(defn- kondo-finding->diagnostic [lines {:keys [message level row col] :as finding}]
+  (let [decrow (max 0 (dec row))
+        deccol (max 0 (dec col))
+        ^String line (when (> (count lines) decrow)
+                       (nth lines decrow))
+        start-char (when (> (count line) deccol)
+                     (.charAt line deccol))
+        expression? (identical? \( start-char)
+        finding (merge {:end-row row :end-col col} finding)]
+    {:range (shared/->range (if expression?
+                              (assoc finding :end-row row :end-col col)
+                              finding))
+    :message (str (string/upper-case (str (first message))) (string/join (rest message)))
+    :severity (case level
+                :error 1
+                :warning 2
+                :info 3)
+    :source "clj-kondo"}))
+
+(def kondo-base-args {:cache true
+                      :cache-dir ".clj-kondo/cache"
+                      :config {:linters (into {} (map #(hash-map % {:level :off})
+                                                      #{:invalid-arity
+                                                        :unused-bindings
+                                                        :unresolved-symbol
+                                                        :unresolved-namespace
+                                                        :unused-namespace}))}})
+
+(defn- run-kondo!
+  ([paths]
+   (kondo/run! (merge kondo-base-args {:lint paths})))
+  ([text lang]
+   (with-in-str text (kondo/run! (merge kondo-base-args {:lint ["-"]
```

Comment:
what this `["-"]` mean? that we won't lint anything?

Reply:
We tell clj-kondo to lint stdin by passing that argument. See: https://github.com/borkdude/clj-kondo.lsp/blob/66a54966d8e5b26643e9b39241ba26c1215f5f9d/server/src/clj_kondo/lsp_server/impl/server.clj#L120-L124